client/additional-methods-setup/ensure-connected-account.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * External dependencies
+ */
+import React, { useContext, useEffect } from 'react';
+import { getHistory, getNewPath } from '@woocommerce/navigation';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import WCPaySettingsContext from 'wcpay/settings/wcpay-settings-context';
+
+const RedirectToConnectPage = () => {
+	useEffect( () => {
+		// temporarily overwriting this so that it appears on the connect page
+		// it will be hidden on page refresh
+		wcpaySettings.errorMessage = __(
+			'Please connect your account before proceeding with onboarding.',
+			'woocommerce-payments'
+		);
+		getHistory().push( getNewPath( {}, '/payments/connect', {} ) );
+	}, [] );
+
+	return null;
+};
+
+const EnsureConnectedAccount = ( { children } ) => {
+	const { accountStatus } = useContext( WCPaySettingsContext );
+
+	if ( Boolean( accountStatus.error ) ) {
```

Comment:
Can `accountStatus` be relied upon to check if an account is connected (and just that)?

[`accountStatus.error` can be false if](https://github.com/Automattic/woocommerce-payments/blob/ccc792aec6a829d61ff79d0268134156d9dc86e3/includes/class-wc-payments-account.php#L151):

```php
if ( empty( $account ) ) {
	// empty means no account. This data should not be used when the account is not connected.
	return [
		'error' => true,
	];
}

if ( ! isset( $account['status'] )
	|| ! isset( $account['payments_enabled'] )
	|| ! isset( $account['deposits_status'] ) ) {
	// return an error if any of the account data is missing.
	return [
		'error' => true,
	];
}
```

This means the redirect will also happen if e.g. `status` and `payments_enabled` are valid, but `deposits_status` is empty.

I am not sure if this is at all possible in a real-life situation but judging by `get_account_status_data()`'s docblock, the method might have only been originally intended for display (maybe for `client/components/account-status/index.js` or `client/account-status-settings/index.js`).

Reply:
I think relying on `get_account_status_data` covers more bases than whether the account is connected or not, so it should be acceptable.

`get_account_status_data` relies from `get_cached_account_data`, which also validates on `is_valid_cached_account`.
The alternative could be `is_stripe_connected`, but that's not part of the `wcpaySettings`. Let me know if you don't think this is acceptable for a quick patch on the release 👍